cmd/cli/plugin/feature/activate.go

```diff
@@ -24,22 +30,89 @@ var FeatureActivateCmd = &cobra.Command{
 }
 
 func init() {
-	FeatureActivateCmd.Flags().StringVarP(&featuregate, "featuregate", "f", "tkg-system", "Activate a Feature gated by a particular FeatureGate")
+	FeatureActivateCmd.Flags().BoolVarP(&userAllowsVoidingWarranty, "--permanentlyVoidAllSupportGuaranteesForThisEnvironment", "v", false, "Allow for the permanent voiding of all support guarantees for this environment. For some features, e.g. experimental features, if a user sets the activation status to one that does not match the default activation, all support guarantees for this environment will be permanently voided.")
 }
 
 func featureActivate(cmd *cobra.Command, args []string) error {
 	featureName := args[0]
-	featureGateClient, err := featuregateclient.NewFeatureGateClient()
+
+	fgClient, err := featuregateclient.New()
 	if err != nil {
-		return fmt.Errorf("couldn't get featureGateRunner: %w", err)
+		return fmt.Errorf("get FeatureGate client: %w", err)
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
 	defer cancel()
 
-	if err := featureGateClient.ActivateFeature(ctx, featureName, featuregate); err != nil {
-		return fmt.Errorf("couldn't activate feature %s: %w", featureName, err)
+	feature, err := fgClient.GetFeature(ctx, featureName)
+	if err != nil {
+		return fmt.Errorf("get Feature %s: %w", featureName, err)
 	}
-	cmd.Printf("Feature %s Activated", featureName)
+
+	displayActivationWarnings(feature)
+
+	gates, err := fgClient.GetFeatureGateList(ctx)
+	_, featRef := featuregateclient.FeatureRefFromGateList(gates, featureName)
+
+	var userAllows *bool
+	if cmd.Flag("userAllowsVoidingWarranty").Changed {
+		userAllows = &userAllowsVoidingWarranty
+	}
+	warrantyVoidAllowed := isVoidingWarrantyAllowed(featRef, feature, userAllows)
```

Comment:
Nope, because `userAllowsVoidingWarranty` is boolean and `userAllows` is pointer to boolean. I need that pointer to see if user set it or not. I don't want to assume `false` if user didn't set it. Why not? Because I need to know if I should ask the user if ze wants to allow voiding warranty. If I assume it was false because of response, I don't ask the user if the user would like to void the warranty.